src/canmatrix/tests/test_canmatrix.py

```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
     assert str(ecu) == "BoardUnit(name='Gateway', comment='with bug')"
 
 
+# Signal (generic functions)
+def test_signal_has_comment(some_signal):
+    comment = "comment01"
+    some_signal.addComment(comment)
+    assert some_signal.comment == comment
+
+
+def test_signal_find_attribute(some_signal):
+    some_signal.addAttribute("attr1", 255)
+    assert some_signal.attribute("attr1") == 255  # check value from 'attributes'
+    assert some_signal.attribute("is_float") is False  # check value from member attribute
+
+
+def test_signal_no_attribute(some_signal):
+    assert some_signal.attribute("wrong") is None  # check default is None
+    assert some_signal.attribute("wrong", default=0) == 0  # check user defined default
+
+
+def test_signal_default_attr_from_db(some_signal):
+    define = canmatrix.canmatrix.Define("INT 0 255")
+    define.defaultValue = 33
+    matrix = canmatrix.canmatrix.CanMatrix(signalDefines={"from_db": define})
+    assert some_signal.attribute("from_db", db=matrix, default=2) == 33
+    assert some_signal.attribute("wrong", db=matrix, default=2) == 2
+
+
+def test_signal_no_duplicate_receiver(some_signal):
+    some_signal.addReceiver("GW01")
+    some_signal.addReceiver("GW01")
+    assert some_signal.receiver == ["GW01"]
+
+
+def test_signal_delete_receiver(some_signal):
+    some_signal.addReceiver("GW01")
+    some_signal.addReceiver("ESP")
+    some_signal.delReceiver("GW01")
+    assert some_signal.receiver == ["ESP"]
+
+
+def test_signal_delete_wrong_receiver_doesnt_raise(some_signal):
+    some_signal.delReceiver("wrong")
+
+
+def test_signal_has_attributes(some_signal):
+    some_signal.addAttribute("attr1", "val1")
+    assert some_signal.attributes == {"attr1": "val1"}
+
+
+def test_signal_delete_attribute(some_signal):
+    some_signal.addAttribute("attr1", "val1")
+    some_signal.delAttribute("attr1")
+    assert some_signal.attributes == {}
+
+
+def test_signal_delete_wrong_attribute_doesnt_raise(some_signal):
+    some_signal.delAttribute("wrong")
+
+
+def test_signal_spn(some_signal):
+    assert some_signal.spn is None
+    some_signal.addAttribute("SPN", 10)
+    assert some_signal.spn == 10
+
+
+def test_signal_set_startbit():
+    signal = canmatrix.canmatrix.Signal(size=16)
+    signal.setStartbit(2)
+    assert signal.startBit == 2
+
+
+def test_signal_set_startbit_conversion():
+    signal = canmatrix.canmatrix.Signal(size=16, is_little_endian=False)
+    signal.setStartbit(20, startLittle=True)
+    assert signal.startBit == 5  # lsb on 20, msb is 20-15 = 5
+    # TODO add test for reversed endianning
+
+
+def test_signal_set_startbit_raise():
+    signal = canmatrix.canmatrix.Signal(size=16, is_little_endian=False)
+    with pytest.raises(Exception):  # TODO change to ValueError?
```

Comment:
I'm still in favor of custom exceptions to isolate from stdlib exceptions. But yes, this is wildly vague.  Separate PR for just that (unless there are other exceptions to tidy similarly).